First release for X-NUCLEO-IKS02A1-Audio (#1)
stm32duino/X-NUCLEO-IKS02A1-Audio: README.md

# X-NUCLEO-IKS02A1 Audio

> Arduino library to support the  IMP34DT05 digital microphone

Reviewer: ```suggestion
Arduino library to support the [IMP34DT05](https://www.st.com/en/mems-and-sensors/imp34dt05.html) digital microphone.
```

## Setting SAI & I2S

> Up to version 2.0 it is necessary to apply the following steps:

Reviewer: ```suggestion
Up to [STM32 core](https://github.com/stm32duino/Arduino_Core_STM32) version 2.0.0 it is necessary to apply the following steps:
```

> Check the path of folder where the file preferences.txt are in Arduino IDE -> Preferences.

Reviewer: ```suggestion
Check the where the Arduino `preferences.txt` file are in `Arduino IDE -> Preferences`.
```

> 	For Example: C:\Users\Duino\AppData\Local\Arduino15

Reviewer: ```suggestion
	For Example: `C:\Users\Duino\AppData\Local\Arduino15` (Windows)
```

> Check the where the Arduino `preferences.txt` file are in `Arduino IDE -> Preferences`.

Reviewer: ```suggestion
Check where the Arduino `preferences.txt` file are in `Arduino IDE -> Preferences`.
```

> Then go to ..\packages\STM32\hardware\stm32\NRversion\cores\arduino\stm32\stm32yyxx_half_conf.h and open the file.

Reviewer: ```suggestion
Then go to `..\packages\STM32\hardware\stm32\<release version>\cores\arduino\stm32\stm32yyxx_half_conf.h` and open the file.
```

> Then go to `..\packages\STM32\hardware\stm32\<release version>\cores\arduino\stm32\stm32yyxx_half_conf.h` and open the file.

Reviewer 2: stm32yyxx_half_conf.h should be stm32yyxx_hal_conf.h

> In the file add this line of code:

Reviewer: ```suggestion
In the file add these lines of code:
```

> 	#if !defined(HAL_I2S_MODULE_DISABLED)
>   		#define HAL_I2S_MODULE_ENABLED
> 	#else
> 	  #undef HAL_I2S_MODULE_ENABLED
> 	#endif
> 
> 	#if !defined(HAL_SAI_MODULE_DISABLED)
> 	  #define HAL_SAI_MODULE_ENABLED
> 	#else
> 	  #undef HAL_SAI_MODULE_ENABLED
> 	#endif

Reviewer: ```suggestion
```C
	#if !defined(HAL_I2S_MODULE_DISABLED)
  		#define HAL_I2S_MODULE_ENABLED
	#else
	  #undef HAL_I2S_MODULE_ENABLED
	#endif

	#if !defined(HAL_SAI_MODULE_DISABLED)
	  #define HAL_SAI_MODULE_ENABLED
	#else
	  #undef HAL_SAI_MODULE_ENABLED
	#endif
\``` 
```
Remove the \, as I don't know how to protect them inside GH suggestion 😉

	
## API

> This library acquire PDM, convert PDM to PCM and store the data on SD Card as WAVE file.

Reviewer: ```suggestion
This library acquires PDM, convert PDM to PCM and store the data on SD Card as WAVE file.
```

### PDM
Before acquire PDM it is required to inizialized it:

> 	PDM.begin();

Reviewer: ```suggestion
`	PDM.begin();`
```

> This library acquires PDM, convert PDM to PCM and store the data on SD Card as WAVE file.

Reviewer: ```suggestion
This library acquires PDM, converts PDM to PCM and stores the data on SD Card as WAVE file.
```

> This library acquires PDM, converts PDM to PCM and stores the data on SD Card as WAVE file.

Reviewer 2: Hi @sarachiodo ,
actually the SD card and wave stuff have been moved on the FP_Examples, so we can say only:

This library acquires PDM and converts PDM to PCM

> Before acquire PDM it is required to initialized it:

Reviewer: ```suggestion
Before acquire PDM it is required to initialize it:
```

> Before acquire PDM it is required to inizialized it:

Reviewer: ```suggestion
Before acquire PDM it is required to initialized it:
```

`   PDM.begin();`

> Thanks  the `PDM` instance, it is possible to start to acquire the data, stop, pause the acquisition and resume.

Reviewer: ```suggestion
Thanks to the `PDM` instance, it is possible to start to acquire the data, stop, pause the acquisition and resume.
```

Stop to acquire the PDM data.
	
	PDM.Stop();

Pause the acquisition.
	
	PDM.Pause();

Resume the acquisition.
	
	PDM.Resume();

To perform a function to process the data (foo in the example), it can call the function below:
	

> 	PDM.onReceive(foo);

Reviewer: ```suggestion
Thanks  the `PDM` instance, it is possible to start to acquire the data, stop, pause the acquisition and resume.

- Start to acquire the PDM data:
	
`	PDM.Record(buffer);`

- Stop to acquire the PDM data:
	
`	PDM.Stop();`

- Pause the acquisition:
	
`	PDM.Pause();`

- Resume the acquisition.
	
`		PDM.Resume();`

- To perform a function to process the data (foo in the example), it can call the function below:
	
`		PDM.onReceive(foo);`
```

### PDM2PCM

Before encoding the data from PDM to PCM, it must be initialized following the procedure below:

> 	pdm2pcm_init();

Reviewer: ```suggestion
`	pdm2pcm_init();`
```

  
  Set the volume of the microphone.

> 	pdm2pcm_volume(volume);
> 	
> 
> 	pdm2pcm(BufIn, BufOut, BLOCK_SIZE);

Reviewer: ```suggestion
```C++
	pdm2pcm_volume(volume);
	

	pdm2pcm(BufIn, BufOut, BLOCK_SIZE);
\```
```
